src/redisvalue.rs

```diff
@@ -8,14 +12,58 @@ pub enum RedisValue {
     BulkRedisString(RedisString),
     StringBuffer(Vec<u8>),
     Integer(i64),
+    Bool(bool),
     Float(f64),
+    BigNumber(String),
+    VerbatimString((String, Vec<u8>)),
     Array(Vec<RedisValue>),
     Error(String),
     StaticError(&'static str),
+    Map(HashMap<RedisValue, RedisValue>),
+    Set(HashSet<RedisValue>),
     Null,
     NoReply, // No reply at all (as opposed to a Null reply)
 }
 
+impl Eq for RedisValue {}
```

Comment:
`the trait `std::cmp::Eq` is not implemented for `f64``

Reply:
Seems like we can not assume that the key will not be f64, though the docs state that Redis will never do it:
https://github.com/antirez/RESP3/blob/master/spec.md#map-type

> Maps can have any other type as field and value, however Redis will use only a subset of the available possibilities. For instance it is very unlikely that Redis commands would return an Array as a key, however Lua scripts and modules will likely be able to do so.

Maybe we can assume that the keys will only be String or i64? @gkorland WDYT?

Reply:
@vityafx after discussion with @gkorland we believe it is good enough implementation. Doubles should never appears as keys inside map, it can only happened with a misbehave module and if it does then we are OK with the limitations.

Reply:
Is it really worth it for something that should not really happened and considered a missus? @gkorland would like your opinion on this one.

Reply:
Pushed a fix, as discussed on slack. Introduce a new enum, `RedisValueKey` which holds only values that are OK to be used as keys or set elements. Panic if Redis return a malformed reply (the reply has a key that can not be used as key like double).